src/main/java/dev/galacticraft/mod/client/gui/screen/ingame/SpaceRaceScreen.java

```diff
@@ -404,4 +297,125 @@ private enum Menu {
         TEAM_COLOR,
         TEAM_FLAG
     }
+
+    private static class SpaceRaceButton extends Button {
+
+        public SpaceRaceButton(Component component, int x, int y, int width, int height, OnPress onPress) {
+            super(x, y, width, height, component, onPress, DEFAULT_NARRATION);
+        }
+
+        @Override
+        protected void renderWidget(GuiGraphics graphics, int mouseX, int mouseY, float delta) {
+            int x = this.getX(), y = this.getY();
+            int backgroundColor = this.isHoveredOrFocused() ? 0xAA1e1e1e : 0xAA000000;
+            int lineColor = this.isHoveredOrFocused() ? 0xFF3c3c3c : 0xFF2d2d2d;
+            graphics.fill(x, y, x + width, y + height, backgroundColor);
+            graphics.renderOutline(x, y, width, height, lineColor);
+            this.renderString(graphics, Minecraft.getInstance().font, 0xFFFFFFFF);
+        }
+    }
+
+    private static class ComingSoonButton extends SpaceRaceButton {
+        private final Component originalMessage;
+
+        public ComingSoonButton(Component component, int x, int y, int width, int height) {
+            super(component, x, y, width, height, button -> {});
+            this.originalMessage = component;
+        }
+
+        @Override
+        protected void renderWidget(GuiGraphics graphics, int mouseX, int mouseY, float delta) {
+            this.setMessage(this.isHoveredOrFocused() ? Component.translatable(Translations.SpaceRace.COMING_SOON) : this.originalMessage);
+            super.renderWidget(graphics, mouseX, mouseY, delta);
+        }
+    }
+
+    private static class CustomizeFlagButton extends AbstractButton {
+        private final Runnable onPress;
+        private final ResourceLocation imageLocation;
+
+        public CustomizeFlagButton(int x, int y, int width, int height, ResourceLocation imageLocation, Runnable onPress) {
+            super(x, y, width, height, Component.translatable(Translations.SpaceRace.CUSTOMIZE_FLAG));
+            this.onPress = onPress;
+            this.imageLocation = imageLocation;
+        }
+
+        @Override
+        public void onPress() {
+            this.onPress.run();
+        }
+
+        @Override
+        protected void renderWidget(GuiGraphics graphics, int mouseX, int mouseY, float delta) {
+            graphics.pose().pushPose();
+            graphics.pose().translate(this.getX(), this.getY(), 0.0f);
+            graphics.pose().scale(2.0f, 2.0f, 2.0f);
+            graphics.blit(this.imageLocation, 0, 0, 0, 0, 0, 48, 32, 48, 32);
+            graphics.pose().popPose();
```

Comment:
Some thing here, but also use the widget's width/height instead of scaling up x2

Reply:
I tried using the width and height arguments to graphics.blit(), but that just makes it repeat the flag texture to fill space instead of scaling it up. Then that made the X/Y arguments also double, so I changed it to use pose translation instead. This was all taken from how vanilla renders items in the inventory (in GuiGraphics).

Reply:
```suggestion
            graphics.blit(this.imageLocation, this.getX(), this.getY(), this.width, this.height, 0, 0, 48, 32, 48, 32);
```